java/code/src/com/redhat/rhn/taskomatic/task/RhnJavaJob.java

```diff
@@ -31,37 +32,45 @@
 import java.util.Arrays;
 
 
-
 /**
  * RhnJavaJob
  */
 public abstract class RhnJavaJob implements RhnJob {
 
+    protected static String defaultPluginName = "File";
     protected Logger log = LogManager.getLogger(getClass());
 
     protected Logger getLogger() {
         return log;
     }
 
-    void enableLogging(TaskoRun run) {
+    protected void enableLogging(TaskoRun run) {
+
+        var loggerName = this.getClass().getName();
+        final var config = ((LoggerContext) LogManager.getContext(false)).getConfiguration();
+        for (var appender : config.getLoggerConfig(loggerName).getAppenders().values()) {
```

Comment:
I would move this complete loop to the end of `excute()` function... or at least call it there too. This would avoid dangling appender file handles between two runs

Reply:
Would you please add this change? I don't quite understand. It seems to me we enable appenders and then remove it again.

Reply:
This is what I was thinking about:
```
diff --git a/java/code/src/com/redhat/rhn/taskomatic/task/RhnJavaJob.java b/java/code/src/com/redhat/rhn/taskomatic/task/RhnJavaJob.java
index 22c87e466c2..3a39cd3a91e 100644
--- a/java/code/src/com/redhat/rhn/taskomatic/task/RhnJavaJob.java
+++ b/java/code/src/com/redhat/rhn/taskomatic/task/RhnJavaJob.java
@@ -47,12 +47,7 @@ public abstract class RhnJavaJob implements RhnJob {
     protected void enableLogging(TaskoRun run) {
 
         var loggerName = this.getClass().getName();
-        final var config = ((LoggerContext) LogManager.getContext(false)).getConfiguration();
-        for (var appender : config.getLoggerConfig(loggerName).getAppenders().values()) {
-            appender.stop();
-            config.getLoggerConfig(loggerName).removeAppender(appender.getName());
-        }
-        Configurator.reconfigure(config);
+        cleanLogging();
 
         var builder = ConfigurationBuilderFactory.newConfigurationBuilder();
 
@@ -73,6 +68,16 @@ public abstract class RhnJavaJob implements RhnJob {
         Configurator.reconfigure(builder.build());
     }
 
+    private void cleanLogging() {
+        var loggerName = this.getClass().getName();
+        final var config = ((LoggerContext) LogManager.getContext(false)).getConfiguration();
+        for (var appender : config.getLoggerConfig(loggerName).getAppenders().values()) {
+            appender.stop();
+            config.getLoggerConfig(loggerName).removeAppender(appender.getName());
+        }
+        Configurator.reconfigure(config);
+    }
+
     /**
      * {@inheritDoc}
      */
@@ -105,6 +110,7 @@ public abstract class RhnJavaJob implements RhnJob {
         finishJob();
         HibernateFactory.commitTransaction();
         HibernateFactory.closeSession();
+        cleanLogging();
     }
 
     /**
```

Reply:
Ah ok, I just didn't realize the logging lives only inside the method. Now it's clear thanks